viewing_party/party.py

```diff
@@ -1,23 +1,188 @@
 # ------------- WAVE 1 --------------------
+def create_movie(movie_title, genre, rating):
+    """
+    create a movie profile, return movie; return None if any of the values is none
+    """
+    #add new movie
+    new_movie = {
+        "title": movie_title,
+        "genre": genre,
+        "rating": rating,
+    }
+    
+    for value in new_movie.values():
+        if value == None:
+            return None
 
-def create_movie(title, genre, rating):
-    pass
+    return new_movie
+
+
+def add_to_watched(user_data, movie):
+    """
+    add watched movies to "watched"
+    """
+    user_data["watched"].append(movie)
+    return user_data
+
+
+def add_to_watchlist(user_data, movie):
+    """
+    add movies to watch to watchlist
+    """
+    user_data["watchlist"].append(movie)
+    return user_data
+
+
+def watch_movie(user_data, movie_title):
+    """
+    moves movie from "watch list" to "watched"
+    """
+
+    copy_of_user_data = dict(user_data)
+    movie_ready_to_move = {}
+    for i in range(len(copy_of_user_data["watchlist"])):
+        if movie_title in copy_of_user_data["watchlist"][i]["title"]:
+            movie_ready_to_move = copy_of_user_data["watchlist"][i]
+            user_data["watchlist"].remove(movie_ready_to_move)
+            user_data["watched"].append(movie_ready_to_move)
+    return user_data       

```

Comment:
Great work in avoiding the dangers of modifying a data structure while iterating by using a ranged loop. Because you made a copy, you could iterate through the copy directly and modify the original data structure.

```python
    for movie in copy_of_user_data["watchlist"]:
        if movie_title == copy_of_user_data["watchlist"]["title"]:
            user_data["watchlist"].remove(movie)
            user_data["watched"].append(movie)
    return user_data       
```